spec/models/reservations_validations_spec.rb

```diff
@@ -60,4 +62,20 @@
       end
     end
   end
+
+  describe "conditionally run validations" do
+    context "with reserve_start_at, reserve_end_at, and reservation_changed? is true" do
```

Comment:
the context this block uses is created outside of this file (in the `setup_reservation` factory)--what do you think about making this context explicit by inserting a `let(:reservation)` that specifically defines the fields used by this condition?

then we can depend on the spec at least having the context we state it has, even if someone updates the factory

(I'm sure this happens a lot but I just happened to think about it here, so up to you)